### PR TITLE
Update with changes on SPA PKCE apps for OIN

### PR DESCRIPTION
**Pull request by IanHakes-okta**
## Description:
- **What's changed?** 
Update the SSO integration guide for OIN to include details about OIDC SPA apps that use PKCE

- **Is this PR related to a Monolith release?**
No

### Resolves:

* [OKTA-389293](https://oktainc.atlassian.net/browse/OKTA-389293)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
 1. Based on the [type of application that you have built](/docs/concepts/oauth-openid/#what-kind-of-client-are-you-building), determine the correct [OAuth 2.0 flow](/docs/concepts/oauth-openid/#recommended-flow-by-application-type) that is required below the OIDC identity layer.
-    >**Note:** For OIDC app integrations intended for publication in the OIN, you must use a Web (server-side) platform with an authorization code flow.
-1. Determine the login redirect URIs on your system. A redirect URI is where Okta sends the authentication response and ID token during the sign-in flow. You can specify more than one URI if required.
+
+    For OIDC applications destined for the OIN, you can create either of the following:
+
+    * A Web application with a dedicated server-side back-end capable of securely storing a Client Secret and exchanging information with an authorization server through trusted back-channel connections. Okta recommends using the Authorization Code flow with an exchange of the client credentials (client ID and Client Secret) for controlling the access between your application and the resource server.
```
Comment (kasturikumthekar-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:16]: consistent casing - either uppercase or lowercase for client ID and client secret.

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
+    * A Single Page Application (SPA), using an Authorization Code flow with a Proof Key for Code Exchange (PKCE) as the recommended method for controlling the access between your application and a resource server.
+
+1. Determine the sign-in redirect URIs on your system. A redirect URI is where Okta sends the authentication response and ID token during the sign-in flow. You can specify more than one URI if required.
```
Comment (kasturikumthekar-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:19]: Incorrect Numbering. The line below is also 1.

**Review by kasturikumthekar-okta — COMMENTED** [review]
Minor changes

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
-    >**Note:** For OIDC app integrations intended for publication in the OIN, you must use a Web (server-side) platform with an authorization code flow.
-1. Determine the login redirect URIs on your system. A redirect URI is where Okta sends the authentication response and ID token during the sign-in flow. You can specify more than one URI if required.
+
+    For OIDC applications destined for the OIN, you can create either of the following:
+
+    * A Web application with a dedicated server-side back-end capable of securely storing a Client Secret and exchanging information with an authorization server through trusted back-channel connections. Okta recommends using the Authorization Code flow with an exchange of the client credentials (client ID and Client Secret) for controlling the access between your application and the resource server.
+    * A Single Page Application (SPA), using an Authorization Code flow with a Proof Key for Code Exchange (PKCE) as the recommended method for controlling the access between your application and a resource server.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:17]:
```suggestion
    * A Single Page Application (SPA), using an Authorization Code flow with a Proof Key for Code Exchange (PKCE), as the recommended method for controlling the access between your application and a resource server.
```

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
 
-Before you create a new OIDC integration in Okta:
-
-<!-- [ian 2020.02.25] this step doesn't matter unless the user can choose something besides the web as a platform
-1. Decide which platform you'll be using for the integration:
-   * A web application is accessed through the browser and can remain running on a server that can store a secret safely.
-   * A native application resides on the end user's device.
-   * A single-page app (SPA) is a web application that is contained on a single web page. All code is retrieved when the page is loaded initially - the page doesn't reload or refresh. A SPA application cannot keep running on a server.
--->
+Now, before you create a new OIDC integration in Okta:
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:9]:
```suggestion
Before you create a new OIDC integration in Okta:
```

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 1 above)
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:16]:
```suggestion
    * A Web application with a dedicated server-side back-end that is capable of securely storing a Client Secret and exchanging information with an authorization server through trusted back-channel connections. Okta recommends using the Authorization Code flow with an exchange of the client credentials (client ID and Client Secret) for controlling the access between your application and the resource server.
```

**Review by christinechoy-okta — COMMENTED** [review]
A few comments, otherwise LGTM!

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 2 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:19]: That's normal markdown - if you look at the output https://github.com/okta/okta-developer-docs/pull/2126/files?short_path=a292798#diff-a292798b6571630f448c8ca2a8504af6ee155663049b6290a3aa34a78b30bbfd - the numbering of the steps is correct. (Markdown processes all lines that start with the number 1 in a sequential count).

No change required.

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 1 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:16]: Changed "client ID" to "Client ID"

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 4 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:9]: I added the "Now," here to differentiate between the multi-tenancy stuff at the top, but looking at it now, it's too informal and obviously doesn't flow the way I intended. Instead I've added h4 headers for the top "Multi-tenancy" and here "Prerequisites"

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 3 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:17]: Changed to:
* A Single Page Application (SPA) that uses an Authorization Code flow with a Proof Key for Code Exchange (PKCE). Okta recommends this method to control the access between your SPA application and a resource server.

**Review by christinechoy-okta — APPROVED** [review]

**Pull request merged**